commit e88d1eb554cfcc6f8c1800403a3efd20a4175c16
Author: Steve Wise <swise@opengridcomputing.com>
Date:   Tue Feb 21 11:21:57 2017 -0800

    rdma_cm: fail iwarp accepts w/o connection params
    
    commit f2625f7db4dd0bbd16a9c7d2950e7621f9aa57ad upstream.
    
    cma_accept_iw() needs to return an error if conn_params is NULL.
    Since this is coming from user space, we can crash.
    
    Reported-by: Shaobo He <shaobo@cs.utah.edu>
    Acked-by: Sean Hefty <sean.hefty@intel.com>
    Signed-off-by: Steve Wise <swise@opengridcomputing.com>
    Signed-off-by: Doug Ledford <dledford@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linux-4.9.47/drivers/infiniband/core/cma.c
===================================================================
--- linux-4.9.47.orig/drivers/infiniband/core/cma.c
+++ linux-4.9.47/drivers/infiniband/core/cma.c
@@ -3511,6 +3511,9 @@ static int cma_accept_ib(struct rdma_id_
 	struct ib_cm_rep_param rep;
 	int ret;
 
+	if (!conn_param)
+		return -EINVAL;
+
 	ret = cma_modify_qp_rtr(id_priv, conn_param);
 	if (ret)
 		goto out;
Index: linux-4.9.47/dummy/rpi_1325_e88d1eb554cfcc6f8c1800403a3efd20a4175c16.txt
===================================================================
--- /dev/null
+++ linux-4.9.47/dummy/rpi_1325_e88d1eb554cfcc6f8c1800403a3efd20a4175c16.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
